discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -486,9 +489,12 @@ def prepare_for_install(
                 static_ips=static_ips_config
             )
         nodes.start_all()
-        self.wait_until_hosts_are_discovered(nodes_count=nodes_count)
+        self.wait_until_hosts_are_discovered(nodes_count=nodes_count, allow_insufficient=True)
         nodes.set_hostnames(self)
-        self.set_host_roles()
+        if nodes_count != 1:
```

Comment:
and if it is 0 ?

Reply:
can we have parameter in class that will say, single_node ? and then just if self.single_node?

Reply:
Is 0 a valid value ?
Do we need an additional parameter, besides the 6 we already set:
master_count
nodes_count
bootstrap_in_place
high_availability_mode
user_managed_networking ?

Reply:
i mean why not to have one and not 5
self.single_node -> will set all others

Reply:
It is another dependency.  Where will it set all others? And if they are not set for some reason?  Just adds more complexity